### ISSUE TITLE
Remove the call to add HSTS middleware from the API project template

### ISSUE THREAD

**Issue by DamianEdwards** [aspnet/AspNetCore#8533]
HSTS is intended for browser scenarios, thus we should remove it from the API project template.

**Comment by blowdart** [comment]
Can we force those templates to be HTTPS only then? :)

**Comment by mkArtakMSFT** [comment]
@DamianEdwards, should we?
> Can we force those templates to be HTTPS only then?

**Comment by cremor** [comment]
What about the new Swagger generation options (#7639)? When those are used to create a project with SwaggerUI, there is a use case for browser access.

**Comment by mkArtakMSFT** [comment]
@glennc and @blowdart waiting on resolution here from regarding the HTTPS enforcement statement above.

**Comment by mkArtakMSFT** [comment]
Talked about this offline and the decision is to cover this as part of the docs, linked as a separate issue above.
